src/core/site.jl

```diff
@@ -61,6 +89,20 @@ struct Site
         roof_squarefeet::Union{Real, Nothing} = nothing,
         min_resil_time_steps::Int=0,
         mg_tech_sizes_equal_grid_sizes::Bool = true,
+        CO2_emissions_reduction_min_pct::Union{Float64, Nothing} = nothing,
```

Comment:
Please change the `Float64`s to `Real`s throughout (unless you explicitly do not want to allow integer inputs)